lib/archethic_web/faucet_rate_limiter.ex

```diff
@@ -46,14 +48,23 @@ defmodule ArchethicWeb.FaucetRateLimiter do
   # Server Call backs
   @impl GenServer
   def init(_) do
+    # Subscribe to PubSub
     schedule_clean()
     {:ok, %{}}
   end
 
+  # Listen to event :new_transaction
+
   @impl GenServer
   def handle_call({:block_status, address}, _from, state) do
+    address =
+      case TransactionChain.fetch_genesis_address_remotely(address) do
+        {:ok, genesis_address} -> genesis_address
+        _ -> address
+      end
+
     reply =
-      if address_state = Map.get(state, address) do
+      if address_state = Map.get(state, String.upcase(address)) do
```

Comment:
What is the purpose of `String.upcase` here ? 
Addresses should be of binary form

Reply:
This is just to store it in the map which is only the facuet_rate_limiter state. When user requests It comes as string so storing in facuet_rate_limiter as address -> some map values , where address is a string key and Upcase to keep consistency

Reply:
map keys can be binary, so the best would be to decode the input in the controller as binary and use it later, so you will remove the case issues

Reply:
Okay makes sense.